enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/BoltInteraction.java

```diff
@@ -0,0 +1,318 @@
+    {
+        return (GraphDatabaseFacade) server.graphDatabaseService();
+    }
+
+    @Override
+    public InternalTransaction startTransactionAsUser( BoltSubject subject ) throws Throwable
+    {
+        AuthSubject authSubject = authManager.login( newBasicAuthToken( subject.username, subject.password ) );
+        return getGraph().beginTransaction( KernelTransaction.Type.explicit, authSubject );
+    }
+
+    @Override
+    public String executeQuery( BoltSubject subject, String call, Map<String,Object> params,
+            Consumer<ResourceIterator<Map<String,Object>>> resultConsumer )
+    {
+        if ( params == null )
+        {
+            params = Collections.emptyMap();
+        }
+        try
+        {
+            subject.client.send( TransportTestUtil.chunk( run( call, params ), pullAll() ) );
+            resultConsumer.accept( collectResults( subject.client ) );
+            return "";
+        }
+        catch (Exception e)
+        {
+            return e.getMessage();
+        }
+    }
+
+    @Override
+    public BoltSubject login( String username, String password ) throws Throwable
+    {
+        BoltSubject subject = subjects.get( username );
+        if ( subject == null )
+        {
+            subject = new BoltSubject( connectionFactory.newInstance(), username, password );
+            subjects.put( username, subject );
+        }
+        else
+        {
+            subject.client.disconnect();
+            subject.client = connectionFactory.newInstance();
+        }
+        subject.client.connect( address ).send( TransportTestUtil.acceptedVersions( 1, 0, 0, 0 ) )
+                .send( TransportTestUtil.chunk( init( "TestClient/1.1",
+                        map( "principal", username, "credentials", password, "scheme", "basic" ) ) ) );
+        assertThat( subject.client, TransportTestUtil.eventuallyRecieves( new byte[]{0, 0, 0, 1} ) );
+        subject.setLoginResult( TransportTestUtil.recvOneMessage( subject.client ) );
+        return subject;
+    }
+
+    @Override
+    public void logout( BoltSubject subject ) throws Exception
+    {
+        subject.client.disconnect();
+        subject.client = connectionFactory.newInstance();
+    }
+
+    @Override
+    public void updateAuthToken( BoltSubject subject, String username, String password )
+    {
+
+    }
+
+    @Override
+    public String nameOf( BoltSubject subject )
+    {
+        return subject.username;
+    }
+
+    @Override
+    public void tearDown() throws Throwable
+    {
+        for ( BoltSubject subject : subjects.values() )
+        {
+            subject.client.disconnect();
+        }
+        subjects.clear();
+        server.graphDatabaseService().shutdown();
+        Path IMPERMANENT_DB_ROLES_PATH = Paths.get( "target/test-data/impermanent-db/data/dbms/roles" );
+        if ( Files.exists( IMPERMANENT_DB_ROLES_PATH ) )
+        {
+            Files.delete( IMPERMANENT_DB_ROLES_PATH );
+        }
```

Comment:
I think this piece of code is originally my fault, by why isn't the roles file remove when a new impermanent-db is created, like the users file?